app/src/utils/logger.ts

```diff
@@ -19,6 +20,41 @@ const logOptions: RemoteLoggerOptions = {
   autoDisableRemoteLoggingIntervalInMinutes,
 }
 
-const BCLogger = new RemoteLogger(logOptions)
+/** Map string (from env) to LogLevel enum; unknown values
+ * fallback to debug */
+const parseEnvLogLevel = (value?: string): LogLevel => {
+  if (!value) return LogLevel.debug
+  switch (value.toLowerCase()) {
+    case 'fatal':
+      return LogLevel.fatal
+    case 'error':
+      return LogLevel.error
+    case 'warn':
+    case 'warning':
+      return LogLevel.warn
+    case 'info':
+      return LogLevel.info
+    case 'trace':
+    case 'test':
+    case 'debug':
+      return LogLevel.debug
+    default:
+      return LogLevel.debug
+  }
+}
+
+/** Factory for creating a new app-specific RemoteLogger instance
+ * If logLevel param omitted, use Config.LOG_LEVEL env mapping.
+ */
+export const createAppLogger = (extraLabels: Record<string, string> = {}, logLevel?: LogLevel): RemoteLogger => {
+  const effectiveLevel = logLevel ?? parseEnvLogLevel(Config.LOG_LEVEL)
+  return new RemoteLogger({
+    ...baseOptions,
+    lokiLabels: { ...baseOptions.lokiLabels, ...extraLabels },
+    logLevel: effectiveLevel,
+  })
+}
```

Comment:
question: does this design allow us to update the log level directly?

I'm thinking of the usecase where we update the log level env variable or inject a new log level into the logger without having to reset the app.